std-bits/table/src/main/java/org/enso/table/data/column/storage/type/StorageType.java

```diff
@@ -1,7 +1,59 @@
 package org.enso.table.data.column.storage.type;
 
+import org.enso.base.polyglot.NumericConverter;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+
 /**
  * Represents an underlying internal storage type that can be mapped to the Value Type that is exposed to users.
  */
 public sealed interface StorageType permits AnyObjectType, BooleanType, DateType, DateTimeType, FloatType, IntegerType, TextType, TimeOfDayType {
+  /**
+   * @return the a common StorageType the two types.
+   */
+  static StorageType findGeneralType(StorageType firstType, StorageType secondType) {
+    if (firstType == null || firstType.equals(secondType)) {
+      return secondType;
+    }
+
+    return switch (firstType) {
+      case IntegerType i -> switch (secondType) {
+        case IntegerType i2 -> i.bits().toInteger() >= i2.bits().toInteger() ? i : i2;
+        case FloatType f -> FloatType.FLOAT_64;
+        default -> AnyObjectType.INSTANCE;
```

Comment:
This is inconsistent with the `cast` logic. It allows to cast e.g. `Boolean` values into `Integer` or `Float` (by turning them to 0/1).

I think we should make this consistent with `cast`.

I'm wondering what is the best way to do this - we already are having more and more places where we unify types and I'm worried the logic may get out of sync. But I don't have any solutions yet. The two similar parts (Enso side unification and Java side unification) are not completely parallel, because the Enso side contains some more types. Maybe we should always unify on Enso side? (not necessarily desirable in the type inferring algorithm for Mixed columns)

Maybe we should just have some tests that ensure that the common parts of these two are consistent with each other?

Reply:
Am removing this in next PR.